File: core/evidence/semantic_decentralization_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def model_semantic_decentralization(interpretations: List[Dict[str, Any]], evidence_count: int) -> Dict[str, Any]:
    dominant = len(interpretations) == 1 and evidence_count < 2
    return {
        "decentralized": not dominant,
        "authority_diffused": len(interpretations) > 1 or evidence_count >= 2,
        "hierarchy_lock_in": False,
        "single_interpretation_dominance": dominant,
    }

File: src/evidence/semanticDecentralizationEngine.ts
/**
 * Converted from Python: core/evidence/semantic_decentralization_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function modelSemanticDecentralization(interpretations: any, evidence_count: any): any {
  var dominant: any = py.and2(py.eq(py.len(interpretations), 1), () => ((evidence_count < 2)));
  return {"decentralized": !py.truthy(dominant), "authority_diffused": py.or2((py.len(interpretations) > 1), () => ((evidence_count >= 2))), "hierarchy_lock_in": false, "single_interpretation_dominance": dominant};
}
